From a879fc8d3e9bdf1ae6289e77314ed2ae28cce272 Mon Sep 17 00:00:00 2001
From: Moritz Fischer <moritz.fischer@ettus.com>
Date: Tue, 11 Jul 2017 10:17:05 -0700
Subject: [PATCH 26/35] ARM: dts: Add devicetree file for NI Project EISCAT SDR

This adds the devicetree file for NI Project EISCAT SDR

Signed-off-by: Moritz Fischer <moritz.fischer@ettus.com>
---
 arch/arm/boot/dts/Makefile           |   1 +
 arch/arm/boot/dts/zynq-ni-eiscat.dts | 163 +++++++++++++++++++++++++++++++++++
 2 files changed, 164 insertions(+)
 create mode 100644 arch/arm/boot/dts/zynq-ni-eiscat.dts

diff --git a/arch/arm/boot/dts/Makefile b/arch/arm/boot/dts/Makefile
index f6eb14290d48..5fd046232be3 100644
--- a/arch/arm/boot/dts/Makefile
+++ b/arch/arm/boot/dts/Makefile
@@ -986,6 +986,7 @@ dtb-$(CONFIG_ARCH_ZYNQ) += \
 	zynq-zed.dtb \
 	zynq-ni-sulfur-rev2.dtb \
 	zynq-ni-sulfur-rev3.dtb \
+	zynq-ni-eiscat.dtb \
 	zynq-zybo.dtb
 dtb-$(CONFIG_MACH_ARMADA_370) += \
 	armada-370-db.dtb \
diff --git a/arch/arm/boot/dts/zynq-ni-eiscat.dts b/arch/arm/boot/dts/zynq-ni-eiscat.dts
new file mode 100644
index 000000000000..14796e8cfa29
--- /dev/null
+++ b/arch/arm/boot/dts/zynq-ni-eiscat.dts
@@ -0,0 +1,163 @@
+/*
+ * National Instruments Ettus Research Project EISCAT SDR
+ *
+ * Copyright (c) 2017 National Instruments Corp.
+ *
+ * SPDX-License-Identifier: (GPL-2.0 OR X11)
+ */
+
+#include "zynq-ni-sulfur-rev3.dts"
+
+&usrpio_i2c0 {
+	db_temp0: hwmon@18 {
+		reg = <0x18>;
+		compatible = "lm95234";
+		#thermal-sensor-cells = <1>;
+	};
+};
+
+&usrpio_i2c1 {
+	db_temp1: hwmon@18 {
+		reg = <0x18>;
+		compatible = "lm95234";
+		#thermal-sensor-cells = <1>;
+	};
+};
+
+&fan0 {
+	cooling-min-state = <0>;
+	cooling-max-state = <13500>;
+};
+
+&fan1 {
+	cooling-min-state = <0>;
+	cooling-max-state = <13500>;
+};
+
+&thermal_zones {
+	eiscat-db0-zone {
+		polling-delay-passive = <1000>;
+		polling-delay = <1000>;
+
+		thermal-sensors = <&db_temp0 0>;
+
+		trips  {
+			db0_alert0: db0-alert0 {
+				temperature = <30000>; /* millicelsius */
+				hysteresis = <2000>;   /* millicelsius */
+				type = "active";
+			};
+
+			db0_alert1: db0-alert1 {
+				temperature = <45000>; /* millicelsius */
+				hysteresis = <2000>;   /* millicelsius */
+				type = "active";
+			};
+
+			db0_alert2: db0-alert2 {
+				temperature = <50000>; /* millicelsius */
+				hysteresis = <2000>;    /* millicelsius */
+				type = "active";
+			};
+
+
+			db0_crit: db0-crit {
+				temperature = <55000>; /* millicelsius */
+				hysteresis = <2000>;    /* millicelsius */
+				type = "critical";
+			};
+
+		};
+
+		cooling-maps {
+			map0 {
+				trip = <&db0_alert0>;
+				cooling-device = <&fan1 0 6000>;
+				contribution = <100>;
+			};
+
+			map1 {
+				trip = <&db0_alert1>;
+				cooling-device = <&fan1 6000 8000>;
+				contribution = <100>;
+			};
+
+			map2 {
+				trip = <&db0_alert2>;
+				cooling-device = <&fan1 8000 12500>;
+				contribution = <100>;
+			};
+
+			map3 {
+				trip = <&db0_crit>;
+				cooling-device = <&fan1 12500 13500>;
+				contribution = <100>;
+			};
+
+		};
+	};
+
+	eiscat-db1-zone {
+		polling-delay-passive = <1000>;
+		polling-delay = <1000>;
+
+		thermal-sensors = <&db_temp1 0>;
+
+		trips  {
+			db1_alert0: db1-alert0 {
+				temperature = <30000>; /* millicelsius */
+				hysteresis = <2000>;   /* millicelsius */
+				type = "active";
+			};
+
+			db1_alert1: db1-alert1 {
+				temperature = <45000>; /* millicelsius */
+				hysteresis = <2000>;   /* millicelsius */
+				type = "active";
+			};
+
+			db1_alert2: db1-alert2 {
+				temperature = <50000>; /* millicelsius */
+				hysteresis = <2000>;    /* millicelsius */
+				type = "active";
+			};
+
+
+			db1_crit: db1-crit {
+				temperature = <55000>; /* millicelsius */
+				hysteresis = <2000>;    /* millicelsius */
+				type = "critical";
+			};
+
+		};
+
+		cooling-maps {
+			map0 {
+				trip = <&db1_alert0>;
+				cooling-device = <&fan1 0 6000>;
+				contribution = <100>;
+			};
+
+			map1 {
+				trip = <&db1_alert1>;
+				cooling-device = <&fan1 6000 8000>;
+				contribution = <100>;
+			};
+
+			map2 {
+				trip = <&db1_alert2>;
+				cooling-device = <&fan1 8000 12500>;
+				contribution = <100>;
+			};
+
+			map3 {
+				trip = <&db1_crit>;
+				cooling-device = <&fan1 12500 13500>;
+				contribution = <100>;
+			};
+
+		};
+	};
+
+
+};
-- 
2.14.2

